mlir/include/mlir/Dialect/Vector/IR/VectorOps.td

```diff
@@ -1363,7 +1363,7 @@ def Vector_TransferReadOp :
                    AffineMapAttr:$permutation_map,
                    AnyType:$padding,
                    Optional<VectorOf<[I1]>>:$mask,
-                   OptionalAttr<BoolArrayAttr>:$in_bounds)>,
+                   BoolArrayAttr:$in_bounds)>,
```

Comment:
Can we use a `DenseBoolArrayAttr`?

Reply:
Ping on this thread? (maybe you missed it)
(It might be something to address in a separate PR as well)

Reply:
I did try this briefly, but hit build errors :( 

Yes, I'd rather leave this for a separate PR.

Reply:
OK let'd do that!

Reply:
I've prototyped this suggestion here: https://github.com/llvm/llvm-project/pull/100334

I haven't updated the tests yet - I'm a bit worried about changing the assembly format from:
```mlir
in_bounds = [true, true, true]
```

to
```mlir
in_bounds =  array<i1: true, true, true>
```

Please have a look and let me know what you think. While I could update the Asm printer for `xfer` Ops, I feel that it might be confusing to "end users". I wanted to see what you think before investing more cycles into this. Thanks!

CC @dcaballe

Reply:
> I've prototyped this suggestion here: https://github.com/llvm/llvm-project/pull/100334

You probably meant another PR?

Reply:
Yup: https://github.com/llvm/llvm-project/pull/100336